target/cheri-common/cheri-helper-utils.h

```diff
@@ -51,15 +99,17 @@ static inline void derive_cap_from_pcc(CPUArchState *env, uint32_t cd,
 #else
     (void)oob_info;
 #endif
-    const cap_register_t *pccp = cheri_get_pcc(env);
+    // Note: we can use a PCC value with an outdated cursor since we are writing
+    // a new address.
```

Comment:
And the address we're writing is guaranteed to be in bounds?

Reply:
No, but in that case we get an oob/untagged value (e.g. auipcc, cgetpccincoffset). I've clarified the comment.

Reply:
Turns out cgetpccincoffset wasn't saving PC. Thanks for reminding me to check!